### PR TITLE
Set reaction order properly

### LINKED ISSUE #249: set_reaction_order in EDB does not have correct behavior
In 'data_model.py' file [lines 865 to 904: function 'set_reaction_order']... 
    
The behavior of this function is not accurate. Reaction orders are net set by whether a species is on the left or right side, reaction order can be different for every component in the reaction, regardless of what side it shows up on. In the current form, you are only able to set reaction order for all left-side components to a single number and all right-side components to a single number.
    
Setting reaction order should be done individually by each component within a single reaction.

### PR DESCRIPTION
**Pull request by dangunter**
## Fixes/Addresses:

#249 

## Changes proposed in this PR:
- Rewrote set_reaction_order
- Rewrote tests
- Cleaned up a bit of repeated magic constants (assign variables and re-use them)
- Added a bit of docs in the how-to section (mostly just a pointer to API docs)

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by aladshaw3 — DISMISSED** [review]
I went ahead and made some necessary changes to the functionality of the function and added some additional tests. I believe this is good to go now.

**Comment by aladshaw3** [comment]
@dangunter Looks like the readthedocs check is failing due to a bunch of warnings (see below):

/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.core'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.data'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.examples'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.tools'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.ui'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.examples.flowsheets.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.examples.flowsheets.full_treatment_train'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.examples.flowsheets.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.examples.flowsheets.lsrro'
/home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.unit_models.rst:7: WARNING: toctree contains reference to nonexisting document 'apidoc/watertap.unit_models.zero_order'
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/docs/checkouts/readthedocs.org/user_builds/watertap/checkouts/266/docs/apidoc/watertap.examples.flowsheets.rst: WARNING: document isn't included in any toctree
done
preparing documents... done

**Comment by aladshaw3** [comment]
@dangunter The doc build is still failing. Warnings state that the table of contents tree is trying to refer to documents that don't exist (see my note above). Are you still working on the docs, or is this some kind of bug?

File: docs/apidoc/watertap.core.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+   :maxdepth: 4
+
+   watertap.core.util
+
+Module contents
+---------------
+
+.. automodule:: watertap.core
+   :members:
+   :show-inheritance:
```
Comment (aladshaw3) [inline docs/apidoc/watertap.core.rst:17]: Why are all these new documents being put into this PR? Was this necessary to get the checks to pass?

File: docs/apidoc/watertap.examples.flowsheets.full_treatment_train.flowsheet_components.chemistry.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+.. automodule:: watertap.examples.flowsheets.full_treatment_train.flowsheet_components.chemistry.test_pretreatment
+   :members:
+   :show-inheritance:
+
+Module contents
+---------------
+
+.. automodule:: watertap.examples.flowsheets.full_treatment_train.flowsheet_components.chemistry
+   :members:
+   :show-inheritance:
```
Comment (aladshaw3) [inline docs/apidoc/watertap.examples.flowsheets.full_treatment_train.flowsheet_components.chemistry.rst:75]: I don't think these should be made into documentation, unless we want them to be part of the proper set of tutorials. 

Some of these files include older test flowsheets that were not even part of the milestone.

**Review by aladshaw3 — APPROVED** [review]
LGTM. New docs were autogenerated to produce apidocs on modules, including flowsheet examples. May require some cleaning in those files later.

**Comment by dangunter** [comment]
@lbianchi-lbl  and @andrewlee94 could I prevail on one of you to approve this.

**Review by lbianchi-lbl — APPROVED** [review]
Not sure I can be good for anything more than a rubberstamp, but nothing jumped to my eye. As I understand, a sizeable portion of the changes in this PR are either apidoc files or formatting (possibly Black?).

**Pull request merged**